pkg/api/handlers/compat/images_build.go

```diff
@@ -221,9 +221,11 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 			var m = []string{}
 			if err := json.Unmarshal([]byte(query.Dockerfile), &m); err != nil {
 				// it's not json, assume just a string
-				m = []string{filepath.Join(contextDirectory, query.Dockerfile)}
+				m = []string{query.Dockerfile}
+			}
+			for _, containerfile := range m {
+				containerFiles = append(containerFiles, filepath.Join(contextDirectory, containerfile))
```

Comment:
Probably want to `path.Clean()` the `containerfile` value before joining, to strip out any weird prefixes like "../../" and friends.